### ISSUE TITLE
Cannot get data of cameras when simulating multiple iris drone

### ISSUE THREAD

**Issue by longvucao** [PX4/sitl_gazebo#991]
Hi, I am trying to simulate multiple iris drone and want to get its camera at the same time. When I worked with a drone, I added the following code to the end of the iris.sdf file to get the camera:
![4](https://github.com/PX4/PX4-SITL_gazebo-classic/assets/122087978/e772af85-2f74-4723-b74f-32851aaa9aa2)

Then I used the file mavros_posix_sitl.launch and it got the image I wanted (you can see it in the picture)
![1](https://github.com/PX4/PX4-SITL_gazebo-classic/assets/122087978/fd6c7f10-204c-4145-9f7d-58c94a4e6d64)
![2](https://github.com/PX4/PX4-SITL_gazebo-classic/assets/122087978/4dae8104-c9c1-4710-8c4f-d2e3575da096)

However when I use the file multi_uav_mavros_sitl.launch to simulate multiple drones (still using the edited iris.sdf file) the desired topic appears but I can't get the image (you can see the difference here picture below)
![3](https://github.com/PX4/PX4-SITL_gazebo-classic/assets/122087978/5a1b420f-0a01-4900-9d25-079a453fb228)

So do you have any ideas to solve this problem?
I really appreciate your help)

**Comment by JohnTGZ** [comment]
You need to find a way to namespace all the different camera topics being published so that they are distinct from each other? i.e. not publishing to the same camera topic.

**Comment by Babylonehy** [comment]
Use group namesapce when you spawn your model into gazebo. You have to make sure your depth camera have <robot_namespace> </robot_namespace>.

**Comment by mun0404** [comment]
@Babylonehy I have tried using group namespace but I am still getting the namespace error.

**Comment by LohitoBurrito** [comment]
@mun0404 did you end up figuring this out, I am really stuck. I am about to create 2 new separate iris drone models with different namespace for the cameras

**Comment by mun0404** [comment]
@LohitoBurrito No, I couldn't figure it out. I am still trying to figure out a solution for it, please let me know if you arrive at a solution.
